internal/services/costmanagement/export_base_resource.go

```diff
@@ -0,0 +1,334 @@
+package costmanagement
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/costmanagement/mgmt/2020-06-01/costmanagement"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/costmanagement/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/costmanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type costManagementExportBaseResource struct{}
+
+func (br costManagementExportBaseResource) arguments(fields map[string]*pluginsdk.Schema) map[string]*pluginsdk.Schema {
+	output := map[string]*pluginsdk.Schema{
+		"active": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"recurrence_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(costmanagement.RecurrenceTypeDaily),
+				string(costmanagement.RecurrenceTypeWeekly),
+				string(costmanagement.RecurrenceTypeMonthly),
+				string(costmanagement.RecurrenceTypeAnnually),
+			}, false),
+		},
+
+		"recurrence_period_start": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.IsRFC3339Time,
+		},
+
+		"recurrence_period_end": {
```

Comment:
should we be explicit this is a time value?

```suggestion
		"recurrence_period_start_time": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ValidateFunc: validation.IsRFC3339Time,
		},

		"recurrence_period_end_time": {
```

Reply:
I see your point, I think it would make more sense to go with `recurrence_period_start_date` in this case.